packages/react/router/README.md

```diff
@@ -0,0 +1,815 @@
+
+Instead of building nested `RouteObject` trees by hand, you compose a **small array of route nodes**. Each helper (`layout`, `index`, `route`, `prefix`) returns a node; you can nest them as needed, and the router turns them into a React Router configuration for you.
+
+```ts
+// src/pages/index.ts
+import { index, route, prefix, layout } from '@equinor/fusion-framework-react-router/routes';
+
+export const pages = layout('./MainLayout.tsx', [
+  // Root index: '/'
+  index('./HomePage.tsx'),
+
+  // Group everything under '/products'
+  prefix('products', [
+    // '/products'
+    index('./ProductsPage.tsx'),
+    // '/products/:id'
+    route(':id', './ProductPage.tsx'),
+  ]),
+]);
+```
+
+### 3. Mount the Fusion Router
+
+```tsx
+// src/Router.tsx
+import { Router } from '@equinor/fusion-framework-react-router';
+import { routes } from './routes';
+
+export default function AppRouter() {
+  // The Router creates a React Router instance wired to Fusion navigation
+  return <Router routes={routes} />;
+}
+```
+
+> [!IMPORTANT]
+> Route components receive **reserved props**: `loaderData`, `actionData`, and `fusion`.  
+> Avoid declaring props with those names yourself – use `RouteComponentProps` types instead.
+
+---
+
+## Best practices
+
+- **Use one `Router` per app**  
+  Keep `@equinor/fusion-framework-react-router` as the *only* place that wires React Router to Fusion. Do not create your own `RouterProvider` instances in parallel.
+
+- **Keep page modules co‑located with route definitions**  
+  Put route components, loaders, actions, and `handle` definitions in the same `src/pages/*.tsx` files. Keep the DSL tree in a small `src/pages/index.ts` module.
+
+- **Use `fusion.context` for shared services**  
+  Put long‑lived services (e.g. `QueryClient`, HTTP API wrapper, `Api` class) on `fusion.context` so they are available in loaders, actions, and components without extra React contexts.
+
+- **Use `handle.route` consistently**  
+  Always describe routes via `handle.route` – even for simple pages – so manifests and docs can rely on a complete route schema.
+
+- **Avoid reserved props on components**  
+  Do not declare your own props named `loaderData`, `actionData`, or `fusion`. Use `RouteComponentProps` for typing instead:
+
+  ```tsx
+  import type { RouteComponentProps } from '@equinor/fusion-framework-react-router';
+
+  export default function MyPage({
+    loaderData,
+    actionData,
+    fusion,
+  }: RouteComponentProps<MyLoaderData, MyActionData>) {
+    // ...
+  }
+  ```
+
+- **Start with data routes, adopt the DSL later**  
+  For most “regular” app development, define a normal React Router **data route tree** (`RouteObject[]` with loaders/actions) and pass it straight to `Router`. The DSL + Vite plugin are powerful but still experimental; they’re best suited for teams that need file‑based routing and manifest generation, and are comfortable helping us iron out edge cases and chunking behaviour.
```

Comment:
All examples above uses the DSL. Maybe this part should be informed higher up in the document.